File: Problem_Solving/Algorithms/Warmup/5_Diagonal_Difference/Solution.py
import math
import os
import random
import re
import sys

def diagonalDifference(arr):
    squareMatrixLen=len(arr)
    total=0

    for i in range(squareMatrixLen):
        total+=arr[i][i]
        total-=arr[i][squareMatrixLen-i-1]

    return abs(total)

if __name__ == '__main__':
    fptr = open(os.environ['OUTPUT_PATH'], 'w')

    n = int(input().strip())

    arr = []

    for _ in range(n):
        arr.append(list(map(int, input().rstrip().split())))

    result = diagonalDifference(arr)

    fptr.write(str(result) + '\n')

    fptr.close()

File: Problem_Solving/Algorithms/Warmup/5_Diagonal_Difference/Solution.php
<?php

function diagonalDifference($arr) {
    $squareMatrixLen=count($arr);
    $leftRightDiagSum=0;
    $rightLeftDiagSum=0;

    for($i=0;$i<$squareMatrixLen;$i++){
        $leftRightDiagSum+=$arr[$i][$i];
        $rightLeftDiagSum+=$arr[$i][$squareMatrixLen-$i-1];
    }

    return abs($leftRightDiagSum-$rightLeftDiagSum);
}

$fptr = fopen(getenv("OUTPUT_PATH"), "w");

$n = intval(trim(fgets(STDIN)));

$arr = array();

for ($i = 0; $i < $n; $i++) {
    $arr_temp = rtrim(fgets(STDIN));

    $arr[] = array_map('intval', preg_split('/ /', $arr_temp, -1, PREG_SPLIT_NO_EMPTY));
}

$result = diagonalDifference($arr);

fwrite($fptr, $result . "\n");

fclose($fptr);

?>
